### PR TITLE
TCVP-2771: Update front-end

### PR DESCRIPTION
**Pull request by saikrishnametpalli-nttdata**
# Description

This PR includes the following proposed change(s):

- [TCVP-2771](https://justice.gov.bc.ca/jira/browse/TCVP-2771)
- Added pagination

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/backend/TrafficCourts/Staff.Service/Models/Disputes/GetAllDisputesParameters.cs [hunk 1]
```diff
@@ -52,19 +52,34 @@ public class GetAllDisputesParameters : IPagable, ISortable
...
 
+    /// <summary>
+    /// The optional page number gives the records from given page
+    /// </summary>
     [FromQuery(Name = "pageNumber")]
     public int? PageNumber { get; set; }
 
+    /// <summary>
+    /// The optional page size sets the record count
+    /// </summary>
```
Comment (KDNeufeld) [inline src/backend/TrafficCourts/Staff.Service/Models/Disputes/GetAllDisputesParameters.cs:82]: Thanks for adding summary comments.

**Review by KDNeufeld — APPROVED** [review]
Lots of changes here. Let's deploy and test it out. Great work.
But try to get your tests passing.

File: src/frontend/staff-portal/src/app/components/paging/paging.component.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<div class="main-pagination">
+    <div class="pagination">
+        <a [class.disabled]="currentPage === 1" (click)="changePage(currentPage - 1)"><<</a>
+        <ng-container *ngFor="let e of [].constructor(totalPages); let i = index">
+            <a [class.active]="i+1 === currentPage" (click)="changePage(i+1)" *ngIf="showPage(i+1)">{{i+1}}</a>
```
Comment (Sharkww) [inline src/frontend/staff-portal/src/app/components/paging/paging.component.html:5]: Why we are not using mat-paginator, it also handled accessibility stuff.
https://material.angular.io/components/paginator/overview

File: src/frontend/staff-portal/src/app/components/table-filters/table-filters.component.html [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<div fxFlex fxFlex.gt-md="60vw" fxLayout="row wrap">
+<div fxFlex fxFlex.gt-md="100" fxLayout="row wrap">
```
Comment (Sharkww) [inline src/frontend/staff-portal/src/app/components/table-filters/table-filters.component.html:1]: `<div fxFlex fxLayout="row wrap">`

**Review by Sharkww — CHANGES_REQUESTED** [review]

File: src/frontend/staff-portal/src/app/components/table-filters/table-filters.component.ts [hunk 4]
```diff
@@ -29,6 +29,7 @@ export class TableFiltersComponent implements OnInit {
       this.tableFilterConfigs[key] = true;
     })
     this.dataFilters = this.tableFilterService.tableFilters[this.tabIndex];
+    this.dataFilters.status = "";
```
[the remark this answers is not in the document]
Comment (saikrishnametpalli-nttdata) [reply src/frontend/staff-portal/src/app/components/table-filters/table-filters.component.ts:32]: Initializing the dropdown for default value

File: src/frontend/staff-portal/src/app/components/paging/paging.component.html (hunk 2 above)
Comment (saikrishnametpalli-nttdata) [reply src/frontend/staff-portal/src/app/components/paging/paging.component.html:5]: Made the custom pagination to align with the BC Gov design standard
https://bcgov.github.io/bootstrap-theme/docs/reference/simple/

File: src/frontend/staff-portal/src/app/components/table-filters/table-filters.component.html (hunk 3 above)
Comment (saikrishnametpalli-nttdata) [reply src/frontend/staff-portal/src/app/components/table-filters/table-filters.component.html:1]: I will check this

File: src/frontend/staff-portal/src/app/components/table-filters/table-filters.component.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (Sharkww) [reply src/frontend/staff-portal/src/app/components/table-filters/table-filters.component.ts:32]: `this.dataFilters.status = this.dataFilters.status ? this.dataFilters.status : "";`

File: src/frontend/staff-portal/src/app/components/paging/paging.component.html [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<div class="main-pagination">
+    <div class="pagination">
+        <a [class.disabled]="currentPage === 1" (click)="changePage(currentPage - 1)"><<</a>
+        <ng-container *ngFor="let e of [].constructor(totalPages); let i = index">
```
Comment (Sharkww) [inline src/frontend/staff-portal/src/app/components/paging/paging.component.html:4]: Can the `[].constructor(totalPages)` move to `.ts` ?

File: src/frontend/staff-portal/src/app/components/paging/paging.component.html (hunk 2 above)
Comment (saikrishnametpalli-nttdata) [reply src/frontend/staff-portal/src/app/components/paging/paging.component.html:5]: I will do this change

File: src/frontend/staff-portal/src/app/components/table-filters/table-filters.component.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (saikrishnametpalli-nttdata) [reply src/frontend/staff-portal/src/app/components/table-filters/table-filters.component.ts:32]: I will make this change

File: src/frontend/staff-portal/src/app/components/paging/paging.component.html (hunk 2 above)
Comment (saikrishnametpalli-nttdata) [reply src/frontend/staff-portal/src/app/components/paging/paging.component.html]: Done

File: src/frontend/staff-portal/src/app/components/table-filters/table-filters.component.html (hunk 3 above)
Comment (saikrishnametpalli-nttdata) [reply src/frontend/staff-portal/src/app/components/table-filters/table-filters.component.html]: Done

File: src/frontend/staff-portal/src/app/components/table-filters/table-filters.component.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (saikrishnametpalli-nttdata) [reply src/frontend/staff-portal/src/app/components/table-filters/table-filters.component.ts]: Done

File: src/frontend/staff-portal/src/app/components/paging/paging.component.html (hunk 5 above)
Comment (saikrishnametpalli-nttdata) [reply src/frontend/staff-portal/src/app/components/paging/paging.component.html]: Done

**Review by Sharkww — APPROVED** [review]
Frontend reviewed

**Pull request merged**